include/nbl/builtin/hlsl/bit.hlsl

```diff
@@ -94,14 +94,30 @@ uint16_t clz(uint64_t N)
 template<>
 uint16_t clz<1>(uint64_t N) { return uint16_t(1u-N&1); }
 
-}
+} //namespace impl
 
 template<typename T>
 uint16_t countl_zero(T n)
 {
     return impl::clz<sizeof(T)*8>(n);
 }
 
+
+namespace impl {
+uint32_t bitfieldInsert(uint32_t base, uint32_t shifted_masked_value, uint32_t lo, uint32_t count)
+{
+    const uint32_t hi = base^lo;
+    return (hi<<count)|shifted_masked_value|lo;
+}
+
+} //namespace impl
+
+uint32_t bitfieldInsert(uint32_t base, uint32_t value, uint32_t offset, uint32_t count)
+{
+    const uint32_t shifted_masked_value = (value&((1u<<count)-1))<<offset;
+    return impl::bitfieldInsert(base,shifted_masked_value,base&((1u<<offset)-1),count);
+}
```

Comment:
these are GLSL equivalents, should go into the `glsl` namespace and `glsl_compat` header

btw does GLSL/HLSL/C++ guarantee to wipe high bits of your `value`? If not we could assert that `value>>count == 0` and not bother with the first masking before `<<offset`

P.S. we need to template this stuff
```cpp
template<Unsigned>
enable_if_t<is_unsigned_v<Unsigned>,Unsigned> bitfieldInsert(Unsigned base, Unsigned value, uint16_t offset, uint16_t count)
```

Reply:
Also because its broken into two functions, its kinda hard to read.